src/Compilers/CSharp/Test/Semantic/Semantics/ConstantTests.cs

```diff
@@ -3541,6 +3541,103 @@ void M()
+                // (7,27): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string S2 = $"{"Level 5"} {"Number 3"}";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$""{""Level 5""} {""Number 3""}""").WithArguments("constant interpolated strings").WithLocation(7, 27),
+                // (8,27): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string S3 = $"{$"{"Spinning Top"}"}";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$""{$""{""Spinning Top""}""}""").WithArguments("constant interpolated strings").WithLocation(8, 27),
+                // (8,30): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string S3 = $"{$"{"Spinning Top"}"}";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$""{""Spinning Top""}""").WithArguments("constant interpolated strings").WithLocation(8, 30),
+                // (9,27): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string S4 = $"Hybrid" + "Testing" + "123";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$""Hybrid""").WithArguments("constant interpolated strings").WithLocation(9, 27),
+                // (10,50): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string S5 = "Hybrid" + "Testing" + $"321";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$""321""").WithArguments("constant interpolated strings").WithLocation(10, 50),
+                // (11,27): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string F1 = $"{S1}";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$""{S1}""").WithArguments("constant interpolated strings").WithLocation(11, 27),
+                // (12,32): error CS8652: The feature 'constant interpolated strings' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //         const string F2 = F1 + $" the {S2}";
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, @"$"" the {S2}""").WithArguments("constant interpolated strings").WithLocation(12, 32));
```

Comment:
Please add some tests for:

* Parameter default values
* When the langversion is too low _and_ the string isn't a constant.

For each of the tests around langversion, our general strategy is to compile twice. Once with the correct minimum langversion to show it works, then once with the max unsupported langversion, to show that it doesn't.

Reply:
Added tests, let me know what you think